src/scos_tekrsa/hardware/tekrsa_sigan.py

```diff
@@ -255,109 +253,113 @@ def acquire_time_domain_samples(
         cal_adjust: bool = True,
     ):
         """Acquire specific number of time-domain IQ samples."""
-        self._capture_time = None
-        if isinstance(num_samples, int) or (
-            isinstance(num_samples, float) and num_samples.is_integer()
-        ):
-            nsamps_req = int(num_samples)  # Requested number of samples
-        else:
-            raise ValueError("Requested number of samples must be an integer.")
-        nskip = int(num_samples_skip)  # Requested number of samples to skip
-        nsamps = nsamps_req + nskip  # Total number of samples to collect
-
-        if cal_adjust:
-            # Get calibration data for acquisition
-            if not (settings.RUNNING_TESTS or settings.MOCK_SIGAN):
-                cal_params = sensor_calibration.calibration_parameters
+        iq_capture_lock = threading.Lock()
```

Comment:
This needs to be removed because it is instantiating a new lock each time and not actually locking because it is not shared. The old sigan_lock = threading.Lock() on line 16 should be restored and line 257 should be changed to with sigan_lock: . I confirmed on .201 that is not locking and breaks calls to the status when they are performed during action execution.